x-pack/plugins/index_lifecycle_management/public/store/middleware/auto_enable_phase.js

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  setPhaseData
+} from '../actions';
+import { getPhaseData } from '../selectors';
+import { PHASE_ENABLED } from '../constants';
+
+const setsPerPhase = {};
+
+export const autoEnablePhase = store => next => action => {
+  const state = store.getState();
+
+  if (action.type === setPhaseData().type) {
+    const { phase } = action.payload;
+    setsPerPhase[phase] = setsPerPhase[phase] || 0;
+    setsPerPhase[phase]++;
```

Comment:
Could you walk me through what this logic does / why it does it?

Reply:
If phase data is being set and this is the first time that is being done, this logic also sets enabled to true for that phase (this is the auto_enable the name of the file refers to).

Reply:
Makes sense! What do you think of breaking that state out into a variable to make this intention clearer?

```js
export const autoEnablePhase = store => next => action => {
  const state = store.getState();

  if (action.type === SET_PHASE_DATA) {
    const { phase } = action.payload;
    
    const isPhaseSetForFirstTime = (
      setsPerPhase[phase] == null
      && !getPhaseData(state, phase, PHASE_ENABLED)
    );

    setsPerPhase[phase] = setsPerPhase[phase] || 0;
    setsPerPhase[phase]++;

    // Auto-enable phases being set for the first time.
    if (isPhaseSetForFirstTime) {
      store.dispatch(setPhaseData(phase, PHASE_ENABLED, true));
    }
  }

  return next(action);
};
```